digitalbazaar/crypto-ld
# Refactor API, extract ed25519 and RSA code into separate repos.

Multiple breaking changes:

* Extract Ed25519 and RSA key pair code into their own separate repos.
* Implement chai-like `.use()` API for installing and specifying individual key types you'll be working with:

```js
    import {Ed25519VerificationKey2018} from 'ed25519-verification-key-2018';
    import {RsaVerificationKey2018} from 'rsa-key-pair';
    import Secp256k1 from 'secp256k1-key-pair';
    import X25519 from 'x25519-key-pair';

    import {CryptoLD} from 'crypto-ld';
    const cryptoLd = new CryptoLD();

    cryptoLd.use(Ed25519VerificationKey2018);
    cryptoLd.use(RsaVerificationKey2018);
    cryptoLd.use(Secp256k1);
    cryptoLd.use(X25519);
```

* Rename `.publicNode()` to `.export()`.
* Remove deprecated `.owner` instance property
* Remove deprecated `.passphrase` instance property, and the `encrypt()` and `decrypt()` methods (these are no longer used).

Addresses issue #40.

* [x] Implement CryptoLD driver class. 
* [x] Implement abstract LDKeyPair class.
* [x] Implement abstract LDVerifierKeyPair class (for signature-related suites like rsa & ed25519).
* [x] Remove ed25519 and RSA specific code (will be added to their own repos).
* [x] Update README, add usage for `signer()` and `verifier()` factory methods.
* [x] Add unit tests for the abstract classes.
* [x] Clean up `package.json` / dependencies.
* [x] Set up Karma browser tests.
